### PR TITLE
Build and installer fixes

### PR DESCRIPTION
**Pull request by mikee47**
**Host**

- Fix host --help option
- Add --loopcount emulator option

**Build**

- Update FixPath to handle spaces in path
- Fix COMPONENT_SEARCH_DIRS for Windows
- Fix appcode object path resolution, ensure object path is relative

**Installer**

- `all` option missing rp2040
- Remove `doc` option from `all`

**CI testing**

- Revise to simplify use by external libraries

### CODE REVIEW

**Comment by slaff** [comment]
@mikee47 Please take a look at the failing CIs.

File: Tools/ci/build.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 set -ex # exit with nonzero exit code if anything fails
 
 # Build times benefit from parallel building
-export MAKE_PARALLEL="make -j3"
+export MAKE_PARALLEL="make -j$(nproc)"
```
Comment (mikee47) [inline Tools/ci/build.sh:5]: @slaff This surely won't work?

Comment (mikee47) [reply Tools/ci/build.sh:5]: My bad. Thanks.

Comment (mikee47) [reply Tools/ci/build.sh:5]: Think I'm going slightly potting switching between linux/windows all the time...

**Comment by mikee47** [comment]
@slaff I wonder if it's time to drop the old toolchains?

**Comment by slaff** [comment]
> @slaff I wonder if it's time to drop the old toolchains?

I ... (hesitation..  mumbling... hesitation) would say yes. Those who cannot install a newer toolchain can still use the LTS version.

**Comment by mikee47** [comment]
> 
> > @slaff I wonder if it's time to drop the old toolchains?
> 
> I ... (hesitation.. mumbling... hesitation) would say yes. Those who cannot install a newer toolchain can still use the LTS version.

Should make development a little less awkward and save some CI time! I'll prep a PR for this.

**Pull request merged**